content/en/blog/2023/native-sidecars/index.md

```diff
@@ -0,0 +1,201 @@
+---
+title: Kubernetes Native Sidecars in Istio
+description: Demoing the new SidecarContainers feature with Istio.
+publishdate: 2023-07-25
+attribution: "John Howard (Google)"
+keywords: [istio,sidecars,kubernetes]
+---
+
+From the beginning of Istio, applications have been augmented with service mesh functionality by using the "sidecar" pattern:
+each Pod gets an additional container running alongside it.
+In Istio's case, this additional container runs the Istio proxy, giving us the classic Istio architecture:
+
+{{< image width="80%"
+    link="/docs/ops/deployment/architecture/arch.svg"
+    caption="Istio Architecture"
+    >}}
+
+While most common in service meshes, the sidecar pattern is used for other contexts as well, such as monitoring agents.
+
+## Sidecar woes
+
+Sidecar containers give a lot of power, but come with some issues.
+While containers within a Pod can share some things, their *lifecycle's* are entirely decoupled.
+To Kubernetes, both of these containers are functionally the same.
+
+However, in Istio they are not the same - the Istio container is required for the primary application container to run,
+and has no value without the primary application container.
+
+This mismatch in expectation leads to a variety of issues:
+* If the application container starts faster than Istio's container, it cannot access the network.
+  This wins the [most +1's](https://github.com/istio/istio/issues/11130) on GitHub by a landslide.
+* If Istio's container shuts down before the application container, it cannot access the network.
+  This often means we shutdown too fast (breaking the network) or too slow (leaving pods running with no purpose).
+* If an application container intentionally exists (typically from usage in a `Job`), Istio's container will still run and keep the pod running - indefinitely.
+  This is also a [top GitHub issue](https://github.com/istio/istio/issues/11659).
+* `InitContainers`, which run before Istio's container starts, cannot access the network.
+
+Countless hours have been spent in the Istio community and beyond to work around these - to limited success.
+
+## Fixing the root cause
+
+While increasingly-complex workarounds in Istio can help alleviate the pain for Istio users, ideally all of this would just work - and not just for Istio.
+Fortunately, the Kubernetes community has been hard at work to address these directly in Kubernetes.
+
+In Kubernetes 1.28, a new feature to add native support for sidecars was merged, closing out over 5 years of ongoing work.
+With this merged, all of our issues can be addressed without workarounds!
+
+While we are on the "GitHub issue hall of fame", [these](https://github.com/kubernetes/kubernetes/issues/25908) [issues](https://github.com/kubernetes/kubernetes/issues/65502) account for #1 and #6 all time issues in Kubernetes - and have finally been closed!
+
+A special thanks goes to the huge group of individuals involved in getting this past the finish line.
```

Comment:
@SergeyKanzhelev  @matthyx - is there a bounded list of people we can give a shout-out too? I wanted to include individuals but didn't want to leave folks off so left it generic. LMK if you have thoughts/preferences/etc

Reply:
I can provide a list of ~10-20 people who was most involved recently. Unfortunately I don't remember everybody involved in early incarnations of this proposal. Maybe it will be better to make this posted separately as a blog post on kubernetes.dev. Let me think about it.

Reply:
Thanks for doing this blog post, I agree with Sergey since the final result is the continuation of so many people work. Let's not upset anyone :-)